ps_featuredproducts.php

```diff
@@ -279,6 +273,15 @@ public function getWidgetVariables($hookName = null, array $configuration = [])
         return false;
     }
 
+    protected function getCacheId($name = null)
+    { 
+        $cacheId = parent::getCacheId($name); 
+        if(isset($this->context->customer->id)){
+            $cacheId .= '|'.$this->context->customer->id;
```

Comment:
please add an (int) casting to $this->context->customer->id =, it can be null.

Reply:
`isset` should return false if customer id is NULL, shouldn't it?

Reply:
yes, (int) casting is also for security